dtmsvr/trans_process.go

```diff
@@ -31,14 +32,14 @@ func (t *TransGlobal) process(branches []TransBranch) error {
 	if t.ExtData != "" {
 		dtmimp.MustUnmarshalString(t.ExtData, &t.Ext)
 	}
-
 	if !t.WaitResult {
-		go func() {
+		go func(ctx context.Context) {
+			t.Context = CopyContext(ctx)
```

Comment:
`t.Context = CopyContext(cdx)` should be put before the `go func(ctx context.Context) {`

Reply:
OK